setup.py

```diff
@@ -38,6 +38,18 @@ class Components:
     )
 
     __version__ = v[: v.rfind("-")].replace("-", ".") if "-" in v else v
+
+    # if version is not a valid PEP440 version, then create a bogus version
+    # that will be used only for development purposes which appends the commit hash
+    if not re.match(r"^\d+(\.\d+)*$", __version__):
```

Comment:
Interesting.
@alexsavulescu : as a reference, could you add example(s) of versions that setuptools complaint about now? 
(Wondering about nightly wheels version)

Reply:
This is what we had for shallow clones:
```
python3 setup.py --version
fatal: No names found, cannot describe anything.
```

Wheels have all tags at hand, so no issue there

Reply:
May be we should revise a logic bit - https://app.circleci.com/pipelines/github/neuronsimulator/nrn/3540/workflows/6bf25dce-4274-48a1-bfd8-a373501696f2/jobs/2070/parallel-runs/0/steps/0-105

Reply:
For reference
```
circleci@ip-172-28-37-244:~/project$ python3 setup.py --version
0.0.dev0+gb3a24603a
circleci@ip-172-28-37-244:~/project$ git describe
9.0.dev-1292-gb3a24603a
```

Reply:
```
(venv) savulesc@bbd-cjngk03:~/Workspace/nrn$ python3 setup.py --version
/home/savulesc/Workspace/nrn/venv/lib/python3.10/site-packages/setuptools/__init__.py:85: _DeprecatedInstaller: setuptools.installer and fetch_build_eggs are deprecated. Requirements should be satisfied by a PEP 517 installer. If you are using pip, you can try `pip install --use-pep517`.
  dist.fetch_build_eggs(dist.setup_requires)
/home/savulesc/Workspace/nrn/venv/lib/python3.10/site-packages/setuptools/dist.py:547: UserWarning: Normalizing '9.0.dev.1291' to '9.0.dev1291'
  warnings.warn(tmpl.format(**locals()))
9.0.dev1291
(venv) savulesc@bbd-cjngk03:~/Workspace/nrn$ git describe
9.0.dev-1291-gdd4910f73
```
🤌

Reply:
for reference: pip version and neuron reported versions are not the same:

```console
+ /gpfs/bbp.cscs.ch/home/kumbhar/nrn/nrn_test_venv_39/bin/python -m pip show neuron-gpu-nightly
Name: NEURON-gpu-nightly
Version: 0.0.dev0+g61f36f493

+ nrniv -c 'print "hello"'
INFO : Using neuron-gpu-nightly Package (Alpha Developer Version)
Warning: no DISPLAY environment variable.
--No graphics will be displayed.
NEURON -- VERSION 9.0.dev-1294-g61f36f493 HEAD (61f36f493) 2023-02-15
```

Reply:
and another: `pip --find-links` doesn't find wheel:

```console
kumbhar@r2i2n2:/gpfs/bbp.cscs.ch/ssd/gitlab_map_jobs/bbpcihpcproj12/J555371/drop$ ls -lrt /gpfs/bbp.cscs.ch/ssd/gitlab_map_jobs/bbpcihpcproj12/J555371/drop
total 233320
-rw-r--r--+ 1 bbpcihpcproj12 bbp 13908033 Feb 15 06:57 NEURON_nightly-0.0.dev0+g61f36f493-cp38-cp38-macosx_10_15_x86_64.whl
-rw-r--r--+ 1 bbpcihpcproj12 bbp 23860466 Feb 15 06:58 NEURON_gpu_nightly-0.0.dev0+g61f36f493-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
-rw-r--r--+ 1 bbpcihpcproj12 bbp 23192774 Feb 15 06:58 NEURON_gpu_nightly-0.0.dev0+g61f36f493-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
-rw-r--r--+ 1 bbpcihpcproj12 bbp 13896528 Feb 15 06:59 NEURON_nightly-0.0.dev0+g61f36f493-cp310-cp310-macosx_10_15_x86_64.whl
-rw-r--r--+ 1 bbpcihpcproj12 bbp 13893148 Feb 15 06:59 NEURON_nightly-0.0.dev0+g61f36f493-cp311-cp311-macosx_10_15_x86_64.whl
-rw-r--r--+ 1 bbpcihpcproj12 bbp 23449278 Feb 15 07:00 NEURON_gpu_nightly-0.0.dev0+g61f36f493-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
-rw-r--r--+ 1 bbpcihpcproj12 bbp 13925452 Feb 15 07:00 NEURON_nightly-0.0.dev0+g61f36f493-cp39-cp39-macosx_10_15_x86_64.whl
-rw-r--r--+ 1 bbpcihpcproj12 bbp 23287044 Feb 15 07:01 NEURON_gpu_nightly-0.0.dev0+g61f36f493-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
-rw-r--r--+ 1 bbpcihpcproj12 bbp 36335017 Feb 15 07:03 nrn-nightly-AMD64.exe
-rw-r--r--+ 1 bbpcihpcproj12 bbp 13284774 Feb 15 07:04 NEURON_nightly-0.0.dev0+g61f36f493-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
-rw-r--r--+ 1 bbpcihpcproj12 bbp 13028039 Feb 15 07:06 NEURON_nightly-0.0.dev0+g61f36f493-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
-rw-r--r--+ 1 bbpcihpcproj12 bbp 13122246 Feb 15 07:06 NEURON_nightly-0.0.dev0+g61f36f493-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
-rw-r--r--+ 1 bbpcihpcproj12 bbp 13696359 Feb 15 07:08 NEURON_nightly-0.0.dev0+g61f36f493-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl

....
+ /gpfs/bbp.cscs.ch/ssd/gitlab_map_jobs/bbpcihpcproj12/J555371/nrn/nrn_test_venv_39/bin/python -m pip install --find-links /gpfs/bbp.cscs.ch/ssd/gitlab_map_jobs/bbpcihpcproj12/J555371/drop neuron-gpu-nightly
[4799](https://bbpgitlab.epfl.ch/hpc/cellular/neuron-gpu-wheel-testing/-/jobs/555371#L4799)Looking in links: /gpfs/bbp.cscs.ch/ssd/gitlab_map_jobs/bbpcihpcproj12/J555371/drop
[4800](https://bbpgitlab.epfl.ch/hpc/cellular/neuron-gpu-wheel-testing/-/jobs/555371#L4800)Collecting neuron-gpu-nightly
[4801](https://bbpgitlab.epfl.ch/hpc/cellular/neuron-gpu-wheel-testing/-/jobs/555371#L4801)  Downloading NEURON_gpu_nightly-9.0.dev1289-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (23.3 MB)
[4802](https://bbpgitlab.epfl.ch/hpc/cellular/neuron-gpu-wheel-testing/-/jobs/555371#L4802)     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 23.3/23.3 MB 90.2 MB/s eta 0:00:00
```